### PR TITLE
Docker image for self-hosted runners

### LINKED ISSUE #861: GitHub Runners
### Description

# VNet-Integrated GitHub Actions Runners for Secure Azure Resource Access

### Summary
@Altinn/team-authorization and @Dialogporten teams run specific tasks via GitHub Actions that require direct communication with Azure resources. These resources are only accessible within an Azure Virtual Network (VNet) and are not exposed to the internet.

- **Dialogporten** requires access to Azure Key Vault for retrieving secrets during their GitHub Actions workflows. They have raised concerns about the security implications of exposing the Key Vault to the internet.
  
- **@Altinn/team-authorization** has set up a PostgreSQL database server and needs to execute specific SQL scripts for user and role configuration post-deployment. This requires direct communication with the database, which cannot be facilitated through ARM APIs but must be done via direct network access.

### Proposed Solution
To address these challenges, we propose setting up VNet-integrated, self-hosted GitHub runners for each Altinn product team. This setup would enable secure, private communication between GitHub Actions and the necessary Azure resources within the VNet, bypassing the need for internet exposure.

### Design Considerations
1. **Isolation and Flexibility**:
   - Each Altinn product team should have its own dedicated VM for running GitHub Actions. This ensures:
     - **DNS Resolution**: DNS Private Links are used to resolve private IPs. Each private link needs to be associated with specific VNets for correct DNS resolution. Using shared runners complicates this, as it requires multiple private links to be associated with a single VNet.
     - **Team Independence**: Altinn product teams should operate independently without needing to interact with each other’s resources through GitHub Actions.
     - **Customization**: Teams can configure their GitHub runners as needed without impacting the setup of other teams.

2. **Deployment Process**:
   - Each team will deploy their own GitHub runner VMs by utilizing a Terraform module, which is to be maintained by @Altinn/team-platform.
   - The Terraform module will include a comprehensive README to guide teams on configuring and creating their GitHub runner instances.

### References
- [Adding a self-hosted runner to a repository](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/adding-self-hosted-runners#adding-a-self-hosted-runner-to-a-repository) - GitHub Documentation

### Security Considerations
* Forks of public repository can potentially run dangerous code on self-hosted runner machine by creating a pull request that executes the code in a workflow.

For more information, see "[About self-hosted runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security-with-public-repositories)."

_No response_

### PR DESCRIPTION
**Pull request by tjololo**
## Description
Docker image that is to be used for self-hosted runners.

## Related Issue(s)
Partially fixes #861 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Review by bengtfredh — APPROVED** [review]
LGTM

**Review by bengtfredh — APPROVED** [review]
LGTM

File: .github/workflows/gh-runner-release.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Release gh-runner
+
+on:
+  push:
+    tags:
+      - "*-ghrunner"
```
Comment (tjololo) [inline .github/workflows/gh-runner-release.yml:6]: Changing from postfix to prefix groups the tags together
```suggestion
      - "ghrunner-*"
```

**Review by bengtfredh — APPROVED** [review]
LGTM

**Pull request merged**